keps/168-2-pending-workloads-visibility/README.md

```diff
@@ -0,0 +1,300 @@
+  - [Alternatives within the proposal](#alternatives-within-the-proposal)
+    - [apiserver-builder library](#apiserver-builder-library)
+<!-- /toc -->
+
+## Summary
+
+This KEP proposes to introduce a new API that allows users to on-demand fetch information about pending workloads in both ClusterQueue and LocalQueue. Users will be able to look up the position of a specific workload in both types of queues and list all pending workloads in a specific queue.
+
+## Motivation
+
+As presented in [KEP#168](https://github.com/kubernetes-sigs/kueue/tree/main/keps/168-pending-workloads-visibility),  there is currently a proposal for a mechanism that supports fetching the order of pending workloads, but it comes with a lot of cons. This proposal addresses all of those problems.
+
+### Cons of the current solution
+
+#### Size of the queue
+
+There are a few scalability concerns. The first one is that the number of fetched pending workloads is limited by the etcd object's size limit. By default, a user is able to fetch only 10 workloads stored at the head of a queue. This number can be increased up to 4000, but comes with a performance loss.
+
+#### Consistency across all LocalQueues
+
+Another scalability drawback is that in a Kueue setup with a lot of LocalQueues it is very likely to hit the Kueue QPS. Assuming Kueue setup with multiple LocalQueues pointing to the same ClusterQueue, Kueue needs to send updates to all LocalQueues to update their status, in order to keep the workload positional information up-to-date. This consumes `QPS` which can lead to blocking other requests. 
+
+#### Expanding API in the future
+
+Moreover, there are some functional issues with the current approach. It does not expose any information about pending workloads but for ```name```, ```namespace```, and position in a queue (by listing workloads in order). Adding new fields would result in a decrease in the potential maximum number of fetched pending workloads, caused by the etcd object's size.
+
+#### Delay
+
+Additionally, in the previous proposal, Kueue updated the most prioritized workloads every 5 seconds. It is configurable, but since computing the most prioritized workloads can be expensive, it cannot be significantly reduced. Users can experience delays, which might not be convenient.
+
+
+### Goals
+
+- Support listing in order all pending workloads in a ClusterQueue, no matter the size of the queue, and without delay,
+- Support listing in order all pending workloads in a LocalQueue, no matter the size of the queue, and without delay,
+- Support fetching information about a specific workload, no matter its position in the queue, and without delay,
+- Provide consitent data across all the LocalQueues without hitting `QPS`.
+
+### Non-Goals
+
+- Provide ETA for a workload,
+- Provide information on whether a workload is admissible,
+- Provide information about the requested resource for a workload,
+- Support pagination.
+
+## Proposal
+
+- Add new API,
```

Comment:
The last point does not seem important enough to mention in this section. 

Also, I would suggest to formulate to capture the main technical aspects of the proposal, without going too much detail like in the "Design Details" section (or something along the lines):

"Add a new Extension API server to expose on-demand endpoints for fetching information about pending workloads."

Then, it may also be worth adding something like:
"The returned information about pending workloads includes all the necessary information relevant for their position in the queue, along with the position itself. There are three such endpoints: (1) to list the pending workloads in ClusterQueue, (2) list the pending workloads in LocalQueue, and (3) get a specific pending workload."